### PR TITLE
API Composition UI implementation

### PR DESCRIPTION
**Pull request by praminda**
### Proposed changes in this pull request

- API Composition UI/UX implementation

### When should this PR be merged

After review

### Follow up actions

- Need to implement list all Composite APIs with pagination

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/web/apps/org.wso2.carbon.apimgt.store/pom.xml [hunk 1]
```diff
@@ -55,6 +55,12 @@
             <groupId>org.wso2.carbon.apimgt</groupId>
             <artifactId>org.wso2.carbon.apimgt.authenticator</artifactId>
         </dependency>
+        <dependency>
```
Comment (harsha89) [inline components/web/apps/org.wso2.carbon.apimgt.store/pom.xml]: why we need publisher depenndency to store?

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java [hunk 2]
```diff
@@ -1089,6 +1089,23 @@ public String getCompositeApiDefinition(String id) throws APIManagementException
 
     /**
      * {@inheritDoc}
+     * <p>Implementation will try to retrieve the summary of API to decide if API exist or not</p>
+     */
+    @Override
+    public boolean isCompositeAPIExist(String apiId) throws APIManagementException {
+        boolean status;
+        try {
+            status = getApiDAO().getCompositeAPISummary(apiId) != null;
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java]: can we separate this out. This statement bit cofusing.

File: components/web/apps/org.wso2.carbon.apimgt.store/pom.xml (hunk 1 above)
Comment (praminda) [inline components/web/apps/org.wso2.carbon.apimgt.store/pom.xml]: This is not publisher app dependency. This is a dependency to publisher common component. We take API resource UI implementation from this component without duplicating the implementation.

File: components/web/apps/org.wso2.carbon.apimgt.store/src/main/fragments/composite-api-implementation/composite-api-implementation.hbs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<div class="add-padding-4x">
+    <div class="form-horizontal">
+        <div class="form-group">
+            <p>Goto Ballerina editor inorder to implement your composition logic</p>
```
Comment (pubudu538) [inline components/web/apps/org.wso2.carbon.apimgt.store/src/main/fragments/composite-api-implementation/composite-api-implementation.hbs]: Shall we correct the typo here? 'Go to the Ballerina editor in order to implement your composition logic'

Comment (praminda) [inline components/web/apps/org.wso2.carbon.apimgt.store/src/main/fragments/composite-api-implementation/composite-api-implementation.hbs]: Fixed in  	29f96be

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java (hunk 2 above)
Comment (praminda) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java]: Fixed in  	29f96be

**Pull request merged**